api/src/main/java/org/pentaho/platform/api/email/IEmailService.java

```diff
@@ -39,11 +42,18 @@ public interface IEmailService {
 
   /**
    * Generates a test email via the specficied email configuration
-   * 
+   *
    * @param emailConfig
    *          the email configuration to use for sending the testing email
    */
   public String sendEmailTest( final IEmailConfiguration emailConfig );
 
   public boolean isValid();
+
+  public IEmailAuthenticationResponse getOAuthToken( final IEmailConfiguration emailConfig ) throws Exception;
+
+  public void sendMailGraphApi( final IEmailConfiguration emailConfig, String accessToken, String message ) throws Exception;
```

Comment:
We shouldn't have a specific function for each protocol we need to follow OOP see https://github.com/pentaho/pentaho-platform/pull/5185/files#r1082008758